libcontainer/console.go

```diff
@@ -11,5 +16,59 @@ type Console interface {
 	Path() string
 
 	// Fd returns the fd for the master of the pty.
-	Fd() uintptr
+	File() *os.File
+}
+
+const (
+	TerminalInfoVersion uint32 = 201610041
+	TerminalInfoType    uint8  = 'T'
+)
+
+// TerminalInfo is the structure which is passed as the non-ancillary data
+// in the sendmsg(2) call when runc is run with --console-socket. It
+// contains some information about the container which the console master fd
+// relates to (to allow for consumers to use a single unix socket to handle
+// multiple containers). This structure will probably move to runtime-spec
+// at some point. But for now it lies in libcontainer.
+type TerminalInfo struct {
+	// Version of the API.
+	Version uint32 `json:"version"`
+
+	// Type of message (future proofing).
+	Type uint8 `json:"type"`
```

Comment:
I don't want to expand it (that would be (a)), I want to simplify it (b).  Still future-proof (because either you recognize `terminal` (or `terminal2`, or whatever) or you don't.  Just like with your current proposal, you either recognize `(T, 201610041)` (or `(T, 201612011)`, or whatever) or you don't.

Re: mixing type and version not being clean.  I'd agree if either had any semantic meaning.  But if they're both opaque magic, I don't see the point of matching a tuple when we could just be matching a single field.